lib/indexer/comparer.rb

```diff
@@ -0,0 +1,131 @@
+# require_relative 'app';
+# require 'pry'
+# Indexer::Comparer.new(
+#   'government',
+#   'government-old-2017-06-21t11:04:56z-00000000-0000-0000-0000-000000000000',
+#   'government-2017-06-21t11:55:51z-00000000-0000-0000-0000-000000000000'
+# ).run
+
+module Indexer
+  class Comparer
+    DEFAULT_FIELDS_TO_IGNORE = ["popularity"]
+    BATCH_SIZE = 50
+
+    def initialize(old_index_name, new_index_name, rejected_fields: DEFAULT_FIELDS_TO_IGNORE, max_keys_to_report: 100)
+      @old_index_name = old_index_name
+      @new_index_name = new_index_name
+      @rejected_fields = rejected_fields
+      @max_keys_to_report = max_keys_to_report
+    end
+
+    def log(msg)
+      puts "#{Time.now}: #{msg}"
+    end
+
+    def client
+      Services.elasticsearch(
+        hosts: search_config.elasticsearch["base_uri"],
+        timeout: 30.0
+      )
+    end
+
+    def read_index(index_name, filter = nil)
+      log "Access index: #{index_name}"
+
+      documents = {}
+
+      search_body = filter || { query: { match_all: {} } }
+
+      log "Starting to load index: #{index_name} - #{search_body.inspect}"
+
+      ScrollEnumerator.new(
+        client: client,
+        index_names: index_name,
+        search_body: search_body,
+        batch_size: BATCH_SIZE
+      ) do |document|
+        document.clone.merge('_source' => consistent_hash(reject_fields(document['_source'])))
+      end.each do |d|
+        key = [d['_id'], d['_type']]
+        documents[key] = d
+      end
+
+      log "Finished loading index: #{index_name}"
+
+      documents
+    end
+
+    def reject_fields(hash)
+      hash.reject{ |k, _| @rejected_fields.include?(k) }
+    end
+
+    def consistent_hash(hash)
+      # sort the object as reasonible then convert to a string and final hash teh result
+      # this allow us to check if to objects match with needing to store the full object
+      hash_str = consistentify(hash).inspect
```

Comment:
Can you just call `#hash` on the document, or are there edge cases you're avoiding here?

This does what I'd expect:

```ruby
irb(main):008:0> {a:1,b:2}.hash
=> 3918977153981370324
irb(main):009:0> {b:2,a:1}.hash
=> 3918977153981370324
```